webapp/lib/controller_view_helper.dart

```diff
@@ -131,3 +131,14 @@ view.TagStyle tagTypeToStyle(model.TagType tagType) {
       return view.TagStyle.None;
   }
 }
+
+Map<String, List<model.SuggestedReply>> _groupRepliesIntoCategories(List<model.SuggestedReply> replies) {
+  Map<String, List<model.SuggestedReply>> result = {};
+  for (model.SuggestedReply reply in replies) {
+    if (!result.containsKey(reply.category)) {
+      result[reply.category] = [];
+    }
+    result[reply.category].add(reply);
```

Comment:
@marianamarasoiu If `reply.category` is either `null` or `""`, then should the reply be added to a `"default"` category instead? Then perhaps there would not need to be all the guard code in `controller.dart` lines 230 - 237

Reply:
Good point, the guard code can happen here, and we also avoid the problem of lists overwritting each other which you identified above. Thanks for the suggestion, I've now moved the null check here.

I considered naming it `"default"` but it feels a bit more awkward than the empty string (for example I would need to check again in the view to make sure the default is at the top in the list). I think empty category feels explicit enough? But let's see how it works in practice and if it's not clear, I'll use a `"default"`-type description.